packages/react-async/src/index.d.ts

```diff
@@ -227,6 +227,8 @@ type AsyncInitialWithout<K extends keyof AsyncInitial<T>, T> =
   | Omit<AsyncRejected<T>, K>
 
 type FetchRun<T> = {
+  run(overrideResource: RequestInfo, overrideInit: (init: RequestInit) => RequestInit): void
```

Comment:
isn't overrideInit allowed to be optional here?

Reply:
I would argue "no", because we can only distinguish in a meaningful way between `overrideResource` and  `overrideInit` by the number of arguments being passed. 
As is is much more common to override init than it is to override the resource, the signature with only one argument is held back for init. (Also, v8 was handling it that way, to no need for more breaking changes)

Reply:
I would argue though, that it could make sense to have `overrideInit` as first parameter and `overrideResource` as the second. I know, `fetch` does it the other way round, but having parameters switch place once you add another one might be a confusing behaviour from a developer perspective.
What do you think @ghengeveld ?

Reply:
This whole construct is too confusing to work with, because we now have an overloaded overloaded function (i.e. changing arguments based on both `defer` and whether or not the first argument is a resource). I'm going to just introduce a new type that takes both init and resource in a single `overrides` value.